consolelabs/mochi-api
# chore: add workers for price alert system

**What does this PR do?**

-  [x] Add `watchCoinPriceChanges` worker
-  [x] Add `watchDMNotifyPriceAlert` worker

**How to test**
1. Run `make migrate-up`
2. Open two seperate terminal sessions and run below cmds:
- `go run cmd/watch-dm-notify-price-alert/main.go`
- `go run cmd/watch-coin-price-change/main.go`
3. Add new record by API to see how this two workers operate
- Create new alert 
[POST] http://localhost:8200/api/v1/defi/price-alert
Sample req: `{
    "user_discord_id": "490895538892439553",
    "symbol": "btc",
    "alert_type": "price_drops_to",
    "frequency": "only_once",
    "value": 24050
}`

- Remove an Alert
[DELETE] http://localhost:8200/api/v1/defi/price-alert
Sample req: 
![image](https://user-images.githubusercontent.com/27359472/220814213-c3ca9d7f-59cb-4fd0-9b94-c0136736cfa2.png)


**Notes:**
- Supported symbols are in this list (BTC, ETH, BNB, SOL, FTM, MAGIC). **Logic for this should be finanally updated soon**
- `alert_type` should be in `(price_reaches, price_rises_above, price_drops_to, change_is_over, change_is_under)`
- Alert `frequency` should be in `(only_once, once_a_day, always)`

Thanks for your effort on this PR :kek:
